### PR TITLE
More search filters and channel videos sorting options

### PR DESCRIPTION
**Pull request by EgidioCaprino**
This PR adds more search filters in addition to the previous one: upload date and duration. It also adds a dropdown button in the videos section of the channel screen, with sorting options: newest (default), oldest and popular

https://github.com/lamarios/clipious/assets/5156120/01c1a274-be87-457e-bbb5-644cac25ed03

### CODE REVIEW

**Comment by lamarios** [comment]
fix #404

**Comment by lamarios** [comment]
Thanks a lot again.  So far this looks pretty good but I feel like the channel video drop down looks quite out of place.  I don't have a good solution to that though. I was thinking change it to a button with a sort icon and just above the  video list, put a label with text style of label.small  showing what is the current sorting.

**Comment by EgidioCaprino** [comment]
Hi @lamarios, thank you for the feedback. Do you have an example?

**Comment by lamarios** [comment]
for the design itself, not really. for the text theming: https://github.com/lamarios/clipious/blob/master/lib/player/views/tv/components/player_controls.dart#L28  and applying it: https://github.com/lamarios/clipious/blob/master/lib/player/views/tv/components/player_controls.dart#L86  (The copy with is not necessary if you don't want to change anything)

File: lib/channels/models/channel_sort_by.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+enum ChannelSortBy {
+  newest,
+  oldest,
+  popular;
+
+  String getLable(AppLocalizations locals) {
```
Comment (lamarios) [inline lib/channels/models/channel_sort_by.dart:8]:
```suggestion
  String getLabel(AppLocalizations locals) {
```

**Review by lamarios — CHANGES_REQUESTED** [review]

**Comment by lamarios** [comment]
Sorry i'll have to cause an annoyance to you, i want to merge my PR that changes most of the states and will create some conflicts for you. It should be easy to fix.

**Comment by lamarios** [comment]
Mad eanother bunch of changes, i suggest you do a rebase on master branch, and also can you set up the pre-commit hooks

```
./submodules/flutter/bin/dart run tools/setup_git_hooks.dart
```
So that formatting rules and analysis are done before every commit

**Comment by lamarios** [comment]
also fixes #166

**Comment by GitHubington** [comment]
It would be nice to have an additional filter to distinguish the upload/broadcast states: pending, live, complete.

**Review by lamarios — APPROVED** [review]

**Comment by lamarios** [comment]
I took the initiative to rebase your branch on master and fix some of the stuff.

**Comment by lamarios** [comment]
> It would be nice to have an additional filter to distinguish the upload/broadcast states: pending, live, complete.

unfortunately this is not part of the invidious capabilities

**Pull request merged**